test/test_notify.rb

```diff
@@ -55,4 +55,23 @@ def test_watchdog_is_enabled
       assert_equal true, SystemdDaemon::Notify.watchdog?
     }
   end
+
+  def test_listen_fds_with_names
+    expected_names = { "bar" => 4, "foo" => 3 }
+    with_env('LISTEN_FDNAMES' => expected_names.keys.reverse.join(':'),
```

Comment:
`.reverse` feels a bit odd. The names should be ordered by their fd number, and this is not explicit. I'd suggest to just hard-code the value and add a comment stating that order is important. e.g.

```
'LISTEN_FDNAMES' => "foo:bar", # ordered by fd
```
```